packages/flutter/lib/src/services/text_formatter.dart

```diff
@@ -11,6 +11,39 @@ import 'package:flutter/foundation.dart';
 import 'text_editing.dart';
 import 'text_input.dart';
 
+/// {@template flutter.services.textFormatter.maxLengthEnforcement}
+/// ### [MaxLengthEnforcement.enforced] versus
+/// [MaxLengthEnforcement.allowComposingTextToFinish]
+///
+/// Both [MaxLengthEnforcement.enforced]
+/// and [MaxLengthEnforcement.allowComposingTextToFinish] make sure the final length
+/// of the text does not exceed the max length specified.
+/// The difference is that [MaxLengthEnforcement.enforced] truncates the text
+/// whenever a new character is inserted into the text field,
+/// while [MaxLengthEnforcement.allowComposingTextToFinish] does not truncate when
+/// the user has unfinished composing text, allowing the "placeholder" text to
+/// temporarily exceed the max length limit, which provides a better
+/// user experience for entering ideographic characters (e.g., CJK characters)
+/// via composing on phonetic keyboards.
+///
+/// However, some input methods (Gboard for example) initiate text composition
+/// even for Latin characters. If the text field should only take
+/// Latin characters or numbers (e.g., verification code),
+/// [MaxLengthEnforcement.allowComposingTextToFinish] is generally a better fit.
+/// {@endtemplate}
```

Comment:
nit: I think without the "max length specified" is out of context. 

```
Seealso:

   * [TextField.maxLengthEnforcement] which is used in conjunction with [TextField.maxLength] to limit the length of user input. [TextField] also provides a character counter to provide visual feedback.
```

Reply:
I think `[TextField.maxLengthEnforcement]` or `[TextField.maxLength]` will not be able to jump to the definition.

Reply:
I think it kinda makes sense to reference an out of scope name in a "see also" section.

Reply:
@LongCatIsLooong Analyzer saids no, with a loop import error, see https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket.appspot.com/8865844988744138048/+/steps/prepare_environment/0/steps/Analyze/0/steps/analyze/0/stdout

Reply:
You don't have to import those packages, it's fine to reference OOS identifiers in the documentation.